core/service/src/lib.rs

```diff
@@ -210,13 +213,19 @@ impl<Components: components::Components> Service<Components> {
 			.select(exit.clone())
 			.then(|_| Ok(())));
 
-		let offchain_workers =  if config.offchain_worker {
-			Some(Arc::new(offchain::OffchainWorkers::new(
-				client.clone(),
-				task_executor.clone(),
-			)))
-		} else {
-			None
+		let offchain_workers = match (config.offchain_worker, client.backend().offchain_storage()) {
```

Comment:
@bkchr any ideas how to avoid calling the deprecated `.backend()` function?

Reply:
No sorry, the idea of @gnunicorn is to pass this here around in the initialization. AFAIK.

Reply:
So I guess the `client_db::new_client` would be responsible for creating the offchain workers too?

We could maybe have a generic API that would accept a bunch of `impl FnMut(&Backend)` to avoid coupling everything together.